compiler/src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -76,8 +76,10 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   }
 
   /** A function type or closure with `implicit`, `erased`, or `given` modifiers */
-  class FunctionWithMods(args: List[Tree], body: Tree, val mods: Modifiers)(implicit @constructorOnly src: SourceFile)
-    extends Function(args, body)
+  class FunctionWithMods(args: List[Tree], body: Tree, val mods: Modifiers, val isErased: List[Boolean])(implicit @constructorOnly src: SourceFile)
+    extends Function(args, body) {
+      assert(args.length == isErased.length)
```

Comment:
I assume that this should only be used if we have erased parameters. Therefore we should also have the following invariant:
```scala
isErased.contains(true)
```

Reply:
This is also used when `mods` are not empty as well, so I don't think we can use that assertion here.